__tests__/run_seed.test.js

```diff
@@ -73,6 +73,103 @@ describe("App testing", () => {
           );
         });
     });
+  });
+  describe("GET: queries", () => {
+    test("Returns a 200 status and accepts a sort-by query, which sorts articles by any valid column, sorting by created_at by default", () => {
+      return request(app)
+        .get(`/api/articles?sort_by=title`)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles[0].title).toBe(
+            "UNCOVERED: catspiracy to bring down democracy"
+          );
+          expect(body.articles[body.articles.length - 1].title).toBe("A");
+        });
+    });
+
+    test("Returns a 200 status and accepts a sort-by query sorting by created_at by default", () => {
+      return request(app)
+        .get(`/api/articles`)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles[0].created_at).toBe("2020-11-03T09:12:00.000Z");
+          expect(body.articles[body.articles.length - 1].created_at).toBe(
+            "2020-06-06T09:10:00.000Z"
+          );
+        });
+    });
+    test("200 : accepts an order query, with default set to descending order", () => {
+      return request(app)
+        .get(`/api/articles?sort_by=title&order=ASC`)
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles[0].title).toBe("A");
+          expect(articles[articles.length - 1].title).toBe(
+            "UNCOVERED: catspiracy to bring down democracy"
+          );
+        })
+        .then(() => {
+          return request(app)
+            .get(`/api/articles?sort_by=title`)
+            .expect(200)
+            .then(({ body }) => {
+              expect(body.articles[0].title).toBe(
+                "UNCOVERED: catspiracy to bring down democracy"
+              );
+              expect(body.articles[body.articles.length - 1].title).toBe("A");
+            });
+        });
+    });
+    test("Responds with a 200 status and accepts a topic query", () => {
+      return request(app)
+        .get(`/api/articles?topic=cats`)
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          const catsArticle = {
+            article_id: 5,
+            title: "UNCOVERED: catspiracy to bring down democracy",
+            topic: "cats",
+            author: "rogersop",
+            comment_count: "2",
+            votes: 0,
+            created_at: "2020-08-03T13:14:00.000Z",
+            article_img_url: expect.any(String),
+          };
+          expect(articles[0]).toEqual(catsArticle);
+        });
+    });
+    test("responds with a 404 error when passed a topic that exists but there are no articles for the passed topic", () => {
+      return request(app)
+        .get(`/api/articles/?topic=paper`)
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Not found!");
+        });
```

Comment:
If the topic exists then surely it can be found? I think maybe perhaps you'd just have an empty array of articles?

Reply:
👍 👍